bot/extensions/clashofcode/events.py

```diff
@@ -0,0 +1,51 @@
+import discord
+from discord.ext import commands
+
+from bot import core
+from bot.config import settings
+
+
+class ClashOfCodeEvents(commands.Cog):
+    """Events for clash of code in discord."""
+
+    def __init__(self, bot: core.DiscordBot):
+        self.bot = bot
+
+    @property
+    def role(self):
+        return self.bot.guild.get_role(settings.coc.role_id)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent):
+        if payload.user_id == self.bot.user.id:
+            return
+
+        if payload.message_id != settings.coc.message_id:
+            return
+
+        if self.role in payload.member.roles:
+            return
+
+        await payload.member.add_roles(self.role)
+        try:
+            await payload.member.send(f"Gave you the **{self.role.name}** role!")
+        except discord.HTTPException:
+            pass
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_remove(self, payload: discord.RawReactionActionEvent):
+        if payload.user_id == self.bot.user.id:
+            return
+
+        if payload.message_id != settings.coc.message_id:
+            return
+
+        member = self.bot.guild.get_member(payload.user_id)
+        if self.role not in member.roles:
+            return
+
+        await member.remove_roles(self.role)
+        try:
+            await member.send(f"Removed your **{self.role.name}** role!")
+        except discord.HTTPException:
+            pass
```

Comment:
I thought we were going to set up the welcome screen where people can select roles?

Reply:
yes we are, avib will do that when he has time